cpp/src/cider/type/data/sqltypes.h

```diff
@@ -277,6 +277,10 @@ class SQLTypeInfo {
       , compression(kENCODING_NONE)
       , comp_param(0)
       , size(get_storage_size()) {}
+  SQLTypeInfo(SQLTypes t, SQLTypes st, bool n, const std::vector<SQLTypeInfo>& c)
```

Comment:
I prefer the API should be like `SQLTypeInfo(SQLTypes t, SQLTypeInfo st, bool n` and update the method `get_subtype() ` correspondingly. Current API kind of mix up the Array and Struct definition?